test/foundry/EspressoTEEVerifier.t.sol

```diff
@@ -3,62 +3,121 @@ pragma solidity ^0.8.20;
         string memory inputFile = string.concat(vm.projectRoot(), quotePath);
         bytes memory invalidQuote = vm.readFileBinary(inputFile);
-        bool success = espressoTEEVerifier.verify(invalidQuote);
+        bool success = espressoTEEVerifier.verify(invalidQuote, reportDataHash);
+        assertEq(success, false);
+    }
+
+    /**
+        Test incorrect report data hash
+    */
+    function testIncorrectReportDataHash() public {
+        vm.startPrank(adminTEE);
+
+        string memory quotePath = "/test/foundry/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        bool success = espressoTEEVerifier.verify(sampleQuote, bytes32(0));
+        assertEq(success, false);
+    }
+
+    function testIncorrectMrEnclave() public {
+        vm.startPrank(adminTEE);
+
+        string memory quotePath = "/test/foundry/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        espressoTEEVerifier = new EspressoTEEVerifier(
+            bytes32(0x51dfe95acffa8a4075b716257c836895af9202a5fd56c8c2208dacb79c659ff1),
+            mrSigner,
+            address(0x6E64769A13617f528a2135692484B681Ee1a7169)
+        );
+        bool success = espressoTEEVerifier.verify(sampleQuote, reportDataHash);
         assertEq(success, false);
     }
+
+    function testIncorrectMrSigner() public {
+        vm.startPrank(adminTEE);
+
+        string memory quotePath = "/test/foundry/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        espressoTEEVerifier = new EspressoTEEVerifier(
+            mrEnclave,
+            bytes32(0x51dfe95acffa8a4075b716257c836895af9202a5fd56c8c2208dacb79c659ff5),
+            address(0x6E64769A13617f528a2135692484B681Ee1a7169)
+        );
+        bool success = espressoTEEVerifier.verify(sampleQuote, reportDataHash);
+        assertEq(success, false);
+    }
+
+    function testSetOwner() public {
+        vm.startPrank(adminTEE);
+
+        address newOwner = address(142);
+        espressoTEEVerifier.setOwner(newOwner);
+        assertEq(espressoTEEVerifier.owner(), newOwner);
+        vm.stopPrank();
+    }
+
+    function testSetMrEnclave() public {
+        vm.startPrank(adminTEE);
+
+        bytes32 newMrEnclave = bytes32(
+            0x1234567890123456789012345678901234567890123456789012345678901234
+        );
+        espressoTEEVerifier.setMrEnclave(newMrEnclave);
+        assertEq(espressoTEEVerifier.mrEnclave(), newMrEnclave);
+        vm.stopPrank();
+    }
+
+    function testSetMrSigner() public {
+        vm.startPrank(adminTEE);
+
+        bytes32 newMrSigner = bytes32(
+            0x1234567890123456789012345678901234567890123456789012345678901234
```

Comment:
nit: I think we can write something like `hex"01"` for dummy bytes32 to make it a bit less noisy.